### ISSUE TITLE
The results are different

### ISSUE THREAD

**Issue by Winnie202** [turandai/gaussian_surfels#64]
I have two environments, one is docker and the other is conda. With the same code and input data, why is the mesh result correct in docker but incorrect in conda.How should I solve this problem
my docker env list: 
Package                     Version            Editable project location
--------------------------- ------------------ -------------------------
absl-py                     2.1.0
addict                      2.4.0
aiofiles                    23.2.1
annotated-types             0.7.0
antlr4-python3-runtime      4.9.3
anyio                       4.8.0
asmk                        0.1
AssimpCy                    2.1.1
asttokens                   3.0.0
attrs                       24.3.0
autocommand                 2.2.2
backports.tarfile           1.2.0
beautifulsoup4              4.12.3
black                       24.10.0
blinker                     1.9.0
bracex                      2.5.post1
build                       1.2.2.post1
certifi                     2023.5.7
charset-normalizer          3.2.0
clarabel                    0.9.0
click                       8.1.8
cloudpickle                 3.1.0
cmake                       3.26.4
colormap                    1.2.0
comm                        0.2.2
ConfigArgParse              1.7
contourpy                   1.3.0
cvxpy                       1.6.0
cycler                      0.12.1
Cython                      3.0.11
dash                        2.18.2
dash-core-components        2.0.0
dash-html-components        2.0.0
dash-table                  5.0.0
decorator                   5.1.1
detectron2                  0.6
diff_gaussian_rasterization 0.0.0
diffusers                   0.32.1
einops                      0.8.0
et_xmlfile                  2.0.0
exceptiongroup              1.2.2
executing                   2.1.0
faiss-gpu                   1.7.3
fastapi                     0.115.6
fastjsonschema              2.21.1
ffmpeg-python               0.2.0
ffmpy                       0.5.0
filelock                    3.12.2
Flask                       3.0.3
fonttools                   4.55.3
freetype-py                 2.5.1
fsspec                      2024.2.0
future                      1.0.0
fvcore                      0.1.5.post20221221
gdown                       5.2.0
geffnet                     1.0.2
GPUtil                      1.4.0
gradio                      4.44.1
gradio_client               1.3.0
grpcio                      1.69.0
h11                         0.14.0
h5py                        3.12.1
httpcore                    1.0.7
httpx                       0.28.1
huggingface-hub             0.27.1
hydra-core                  1.3.2
idna                        3.4
imageio                     2.36.1
importlib_metadata          8.5.0
importlib_resources         6.5.2
inflect                     7.3.1
iopath                      0.1.10
ipython                     8.18.1
ipywidgets                  8.1.5
itsdangerous                2.2.0
jaraco.collections          5.1.0
jaraco.context              5.3.0
jaraco.functools            4.0.1
jaraco.text                 3.12.1
jedi                        0.19.2
Jinja2                      3.1.2
joblib                      1.4.2
jsonschema                  4.23.0
jsonschema-specifications   2024.10.1
jupyter_core                5.7.2
jupyterlab_widgets          3.0.13
kapture                     1.1.10
kapture-localization        1.1.10
kiwisolver                  1.4.7
kornia                      0.7.4
kornia_rs                   0.1.8
latextable                  1.0.1
lazy_loader                 0.4
lightglue                   0.0
line_profiler               4.2.0
lit                         16.0.6
llvmlite                    0.43.0
loguru                      0.7.3
Mako                        1.3.8
Markdown                    3.7
markdown-it-py              3.0.0
MarkupSafe                  2.1.3
matplotlib                  3.9.4
matplotlib-inline           0.1.7
mdurl                       0.1.2
mediapy                     1.2.2
more-itertools              10.3.0
mpmath                      1.3.0
mypy-extensions             1.0.0
nbformat                    5.10.4
nest-asyncio                1.6.0
networkx                    3.1
numba                       0.60.0
numpy                       1.26.4
numpy-quaternion            2023.0.4
nvidia-cublas-cu11          11.11.3.6
nvidia-cuda-cupti-cu11      11.8.87
nvidia-cuda-nvrtc-cu11      11.8.89
nvidia-cuda-runtime-cu11    11.8.89
nvidia-cudnn-cu11           9.1.0.70
nvidia-cufft-cu11           10.9.0.58
nvidia-curand-cu11          10.3.0.86
nvidia-cusolver-cu11        11.4.1.48
nvidia-cusparse-cu11        11.7.5.86
nvidia-ml-py                12.570.86
nvidia-nccl-cu11            2.21.5
nvidia-nvtx-cu11            11.8.86
nvitop                      1.4.2
omegaconf                   2.3.0
open3d                      0.18.0
opencv-contrib-python       4.11.0.86
opencv-python               4.10.0.84
openpyxl                    3.1.5
orjson                      3.10.14
osqp                        0.6.7.post3
packaging                   24.2
pandas                      2.2.3
parso                       0.8.4
pathspec                    0.12.1
pexpect                     4.9.0
piexif                      1.1.3
Pillow                      10.0.0
pip                         24.3.1
platformdirs                4.3.6
plotly                      5.24.1
plyfile                     1.1
pooch                       1.8.2
portalocker                 3.1.1
prompt_toolkit              3.0.48
protobuf                    5.29.3
psutil                      6.1.1
ptyprocess                  0.7.0
pure_eval                   0.2.3
pyaml                       25.1.0
pycocotools                 2.0.8
pycolmap                    0.4.0
pycuda                      2024.1.2
pydantic                    2.10.5
pydantic_core               2.27.2
pydub                       0.25.1
pyglet                      2.0.20
Pygments                    2.19.1
pymeshfix                   0.17.0
pymeshlab                   2023.12.post2
PyOpenGL                    3.1.0
pyparsing                   3.2.1
pyproject_hooks             1.2.0
pyquaternion                0.9.9
pyrender                    0.1.45
pysdf                       0.1.9
PySocks                     1.7.1
python-dateutil             2.9.0.post0
python-multipart            0.0.20
pytools                     2024.1.14
pytorch3d                   0.7.2              /home/pytorch3d
pytz                        2024.2
pyvista                     0.44.2
PyYAML                      6.0.2
qdldl                       0.1.7.post5
referencing                 0.35.1
regex                       2024.11.6
requests                    2.31.0
retrying                    1.3.4
rich                        13.9.4
roma                        1.5.1
rpds-py                     0.22.3
ruff                        0.9.1
safetensors                 0.5.2
scikit-image                0.24.0
scikit-learn                1.6.0
scipy                       1.13.1
scooby                      0.10.0
scs                         3.2.7.post2
semantic-version            2.10.0
setuptools                  75.6.0
shapely                     2.0.6
shellingham                 1.5.4
simple_knn                  0.0.0
six                         1.17.0
sniffio                     1.3.1
soupsieve                   2.6
stack-data                  0.6.3
starlette                   0.41.3
sympy                       1.13.1
tabulate                    0.9.0
tenacity                    9.0.0
tensorboard                 2.18.0
tensorboard-data-server     0.7.2
termcolor                   2.5.0
texttable                   1.7.0
threadpoolctl               3.5.0
tifffile                    2024.8.30
tokenizers                  0.21.0
tomli                       2.0.1
tomlkit                     0.12.0
torch                       2.1.1+cu118
torchaudio                  2.1.1+cu118
torchvision                 0.16.1+cu118
tqdm                        4.67.1
traitlets                   5.14.3
transformers                4.47.1
trimesh                     4.5.3
triton                      2.1.0
typeguard                   4.3.0
typer                       0.15.1
typing_extensions           4.12.2
tzdata                      2024.2
urllib3                     2.0.3
uvicorn                     0.34.0
vtk                         9.3.1
wcmatch                     10.0
wcwidth                     0.2.13
websockets                  12.0
Werkzeug                    3.0.6
wheel                       0.45.1
widgetsnbextension          4.0.13
yacs                        0.1.8
zipp                        3.21.0

the result is ![Image](https://github.com/user-attachments/assets/4d97d281-f498-4541-87fa-98b810b14e50)

my conda env list:
Package                     Version
--------------------------- ------------------
absl-py                     2.2.2
addict                      2.4.0
asttokens                   3.0.0
attrs                       25.3.0
backcall                    0.2.0
blinker                     1.8.2
cachetools                  5.5.2
certifi                     2025.4.26
charset-normalizer          3.4.1
click                       8.1.8
comm                        0.2.2
ConfigArgParse              1.7
contourpy                   1.1.1
cycler                      0.12.1
dash                        3.0.4
decorator                   5.2.1
diff_gaussian_rasterization 0.0.0
executing                   2.2.0
fastjsonschema              2.21.1
Flask                       3.0.3
fonttools                   4.57.0
fvcore                      0.1.5.post20221221
google-auth                 2.39.0
google-auth-oauthlib        1.0.0
grpcio                      1.70.0
idna                        3.10
imageio                     2.35.1
importlib_metadata          8.5.0
importlib_resources         6.4.5
iopath                      0.1.10
ipython                     8.12.3
ipywidgets                  8.1.6
itsdangerous                2.2.0
jedi                        0.19.2
Jinja2                      3.1.6
joblib                      1.4.2
jsonschema                  4.23.0
jsonschema-specifications   2023.12.1
jupyter_core                5.7.2
jupyterlab_widgets          3.0.14
kiwisolver                  1.4.7
lazy_loader                 0.4
llvmlite                    0.41.1
Mako                        1.3.10
Markdown                    3.7
MarkupSafe                  2.1.5
matplotlib                  3.7.5
matplotlib-inline           0.1.7
mediapy                     1.2.3
narwhals                    1.37.1
nbformat                    5.10.4
nest-asyncio                1.6.0
networkx                    3.1
numba                       0.58.1
numpy                       1.24.4
oauthlib                    3.2.2
open3d                      0.18.0
opencv-python               4.10.0.84
packaging                   25.0
pandas                      2.0.3
parso                       0.8.4
pexpect                     4.9.0
pickleshare                 0.7.5
pillow                      10.4.0
pip                         24.2
pkgutil_resolve_name        1.3.10
platformdirs                4.3.6
plotly                      6.0.1
plyfile                     1.0.3
portalocker                 3.0.0
prompt_toolkit              3.0.51
protobuf                    5.29.4
ptyprocess                  0.7.0
pure_eval                   0.2.3
pyasn1                      0.6.1
pyasn1_modules              0.4.2
pycuda                      2025.1
Pygments                    2.19.1
pymeshlab                   2023.12.post2
pyparsing                   3.1.4
pyquaternion                0.9.9
python-dateutil             2.9.0.post0
pytools                     2024.1.14
pytorch3d                   0.7.4
pytz                        2025.2
PyWavelets                  1.4.1
PyYAML                      6.0.2
referencing                 0.35.1
requests                    2.32.3
requests-oauthlib           2.0.0
retrying                    1.3.4
rpds-py                     0.20.1
rsa                         4.9.1
scikit-image                0.21.0
scikit-learn                1.3.2
scipy                       1.10.1
setuptools                  75.1.0
shapely                     2.0.7
simple_knn                  0.0.0
six                         1.17.0
stack-data                  0.6.3
tabulate                    0.9.0
tensorboard                 2.14.0
tensorboard-data-server     0.7.2
termcolor                   2.4.0
threadpoolctl               3.5.0
tifffile                    2023.7.10
torch                       1.12.1+cu116
torchaudio                  0.12.1+cu116
torchvision                 0.13.1+cu116
tqdm                        4.67.1
traitlets                   5.14.3
trimesh                     4.6.8
typing_extensions           4.13.2
tzdata                      2025.2
urllib3                     2.2.3
wcwidth                     0.2.13
Werkzeug                    3.0.6
wheel                       0.44.0
widgetsnbextension          4.0.14
yacs                        0.1.8
zipp                        3.20.2

the result is 

![Image](https://github.com/user-attachments/assets/3164abf2-cefb-4dbd-9bf1-994fc7a8f39d)

**Comment by turandai** [comment]
Hi, the second looks bad. Did the training process looks normal? I am not sure if it is a problem of training of meshing.

**Comment by Winnie202** [comment]
> Hi, the second looks bad. Did the training process looks normal? I am not sure if it is a problem of training of meshing.

yes, the training run was successful. The visualization in the process is like the picture. When the same code is used in different environments, the first result is normal, but the result shown in the above picture appears in the second environment. Where is the problem

![Image](https://github.com/user-attachments/assets/3da171a6-4dd9-4195-85dc-c931200b85aa)
